pkg/liqo-controller-manager/webhooks/shadowpod/spdescription_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shadowpod
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	vkv1alpha1 "github.com/liqotech/liqo/apis/virtualkubelet/v1alpha1"
+)
+
+var _ = Describe("ShadowPod Description", func() {
+	var (
+		spValidator       *Validator
+		err               error
+		errTest           error
+		shadowPod         *vkv1alpha1.ShadowPod
+		spDescription     *Description
+		spDescriptionTest *Description
+		spNamespacedName  types.NamespacedName
+		fakeNamespace     *corev1.Namespace
+		peeringInfo       *peeringInfo
+		fakeClient        client.Client
+		quota             *corev1.ResourceList
+		containers        []containerResource
+		initContainers    []containerResource
+	)
+
+	BeforeEach(func() {
+
+		fakeNamespace = forgeNamespaceWithClusterID(clusterID)
+
+		fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(fakeNamespace).Build()
+
+		spValidator = webhook.Admission{Handler: NewValidator(fakeClient, false)}.Handler.(*Validator)
+
+		peeringInfo = createPeeringInfo(*clusterIdentity, *resourceQuota)
+
+		spNamespacedName = types.NamespacedName{Name: testShadowPodName, Namespace: testNamespace}
+
+		Expect(spValidator.InjectDecoder(decoder)).To(Succeed())
+	})
+
+	Describe("Get or Create a ShadowPod Description", func() {
+		JustBeforeEach(func() {
+			containers = nil
+			containers = append(containers, containerResource{cpu: 100, memory: 100})
+			spDescription, err = peeringInfo.getOrCreateShadowPodDescription(ctx, spValidator.client, forgeShadowPodWithResourceLimits(containers, nil))
+		})
+
+		When("The ShadowPod Description does not exist", func() {
+			BeforeEach(func() {
+				spDescription = nil
+				err = nil
+				delete(peeringInfo.shadowPods, testShadowPodName)
+			})
+			It("should return a new one without errors", func() {
+				Expect(spDescription).ToNot(BeNil())
+				Expect(spDescription.running).To(BeTrue())
+				Expect(err).To(BeNil())
```

Comment:
```suggestion
				Expect(err).ToNot(HaveOccurred())
```

Nit. (same for all occurrences)